src/Window.cpp

```diff
@@ -396,6 +380,7 @@ void Window::onMazeFileComboBoxChanged(QString path) {
         return;
     }
     updateMazeAndPath(maze, path);
+    stats->resetAll();
```

Comment:
Conceptually, I think the stats should be reset whenever the run output is cleared. Basically, right before another algorithm is about to run. Is there any specific reason to *also* clear it here, when the maze changes?